ExcelWriter.py

```diff
@@ -1,130 +1,228 @@
-            self.maxCol = []
-            self.indentFormat = {}
+            self.entries=[]          
             self.name=name
-            self.writtenCells = {}
-            
-    def Add_Header(self, headerLines):
-        row = 0;
-        maxCol = 0;
-        for header in headerLines:
-            col = 0
-            for headerLine in header:
-                self.Write_String(row, col, headerLine)
-                col = col + 1
-                maxCol = max(col, maxCol)
-        self.currentCol = maxCol
-        
+            self.fontNames=[]
+            self.borderFormat = {}            
+            self.mergeable = set()
     
-    def Write_String(self, row, col, stringVal, formatValue):
-        # by putting chr(2) at the front of the string, google docs will not remove
-        # the leading spaces
-        tabString = " " * 4
-        stringVal = stringVal.replace("\t", tabString)
-        self.Inc_Max_Col(stringVal)
+    # fontNames lets us apply different fonts per column
+    # mergeable says whether we should merge all the rows to the 
+    # bottom of this group (we use it for names, all the rows
+    # associated with a name are merged)        
+    def Add_Header(self, headerLines, fontNames, mergeable):
+        self.entries.append(headerLines)
+        self.fontNames=fontNames
+        self.mergeable.update(mergeable)
         
-
-        stringVal = chr(2) + stringVal;
-        stringVal = stringVal.strip()
-        print(stringVal)
-        # storing it in col/row because we are going to walk down the columns & merge the cells
-        if (col not in self.writtenCells):
-            self.writtenCells[col] = []       
-        self.writtenCells[col].append((row, stringVal, formatValue))
-        self.worksheet.write_string(row, col, stringVal, formatValue)
-            
-    def Get_Indent_Format(self, indentSize):
-        if (indentSize not in self.indentFormat):
-            temp_format = self.workbook.add_format({'font_name':'Courier New'})
-            temp_format.set_font_size(11)
-            #temp_format.set_text_wrap()
-            #temp_format.set_shrink()
-            temp_format.set_align('top')
-            temp_format.set_indent(indentSize)
-            self.indentFormat[indentSize] = temp_format
-        return self.indentFormat[indentSize]
+    # Not the most generic name, but the needs to be called when we add the column zero block (student name for us).
+    # This lets us start the next list of lists
+    def Next_Student(self, studentName):
+        self.entries.append([])
+        self.Add_String(studentName, 0)
     
-    def Inc_Row(self):
-        self.currentRow = self.endRow + 1
-        self.endRow = self.currentRow
-        self.currentCol = 0
+    #rip out the tabs & the newlines    
+    def String_Manip(self, stringVal, tabSize):
+        tabString = " " * tabSize 
+        stringVal = stringVal.replace("\t", tabString)
+        stringVal = stringVal.rstrip();
```

Comment:
I see your C-derived muscle memory for terminating semi-colons is as strong as mine ;)